### PR TITLE
Updated Curio to add a version number, OS requirement, and SHA256 value

### PR DESCRIPTION
**Pull request by chrisfinazzo**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

**Comment by chrisfinazzo** [comment]
#54811 should be updated accordingly.

File: Casks/curio.rb [hunk 1]
```diff
@@ -1,11 +1,13 @@
 cask 'curio' do
-  version :latest
-  sha256 :no_check
+  version '12.2.1'
+  sha256 'bda3b8a787b89882d5e8850826bfe1ffad3c0f703b67ae8da10015cc3ec6ce40'
 
   url 'https://zengobi.com/downloads/Curio.zip'
```
Comment (vitorgalvao) [inline Casks/curio.rb]:
```suggestion
  url "https://www.zengobi.com/downloads/Curio#{version.no_dots}.zip"
```

File: Casks/curio.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
 cask 'curio' do
-  version :latest
-  sha256 :no_check
+  version '12.2.1'
```
Comment (vitorgalvao) [inline Casks/curio.rb]:
```suggestion
  version '12.02.10'
```

**Review by vitorgalvao — CHANGES_REQUESTED** [review]
If you give maintainers the option to edit directly when submitting a PR, we can fix and merge things faster.

**Comment by vitorgalvao** [comment]
@chrisfinazzo [Same question](https://github.com/Homebrew/homebrew-cask/issues/54811#issuecomment-438279245): how did you find the issue ([why I ask](https://github.com/Homebrew/homebrew-cask/issues/54811#issuecomment-438342317)).

File: Casks/curio.rb (hunk 2 above)
Comment (chrisfinazzo) [reply Casks/curio.rb]: 12.2.1 is the version hosted at the URL provided. How did you determine this? Current versions are listed [here](https://zengobi.com/support/). Not sure how this resolves.

Comment (vitorgalvao) [reply Casks/curio.rb]: This is taken from the `appcast`. We don’t really care about the version being exact. We care more about if being easily interpolated.

**Pull request merged**